src/BenchmarkDotNet/Portability/RuntimeInformation.cs

```diff
@@ -67,10 +72,19 @@ private static bool IsAotMethod()
 
         internal static string GetArchitecture() => GetCurrentPlatform().ToString();
 
+#if NET6_0_OR_GREATER
+        [System.Runtime.Versioning.SupportedOSPlatformGuard("windows")]
+#endif
         internal static bool IsWindows() => IsOSPlatform(OSPlatform.Windows);
 
+#if NET6_0_OR_GREATER
+        [System.Runtime.Versioning.SupportedOSPlatformGuard("linux")]
+#endif
         internal static bool IsLinux() => IsOSPlatform(OSPlatform.Linux);
 
+#if NET6_0_OR_GREATER
+        [System.Runtime.Versioning.SupportedOSPlatformGuard("osx")]
+#endif
         internal static bool IsMacOSX() => IsOSPlatform(OSPlatform.OSX);
 
         internal static bool IsAndroid() => Type.GetType("Java.Lang.Object, Mono.Android") != null;
```

Comment:
Are these things still valid on net6.0? I think the assembly name changed for iOS for sure and maybe Android also has different things?

There is also the `OperatingSystem.IsAndroid()` that should be used on that.

Reply:
They are valid, `OperatingSystem.IsXX` is just newer and faster API.